### PR TITLE
Add `alert.json` to VD E2E test report

### PR DESCRIPTION
**Pull request by MARCOSD4**
# Description

This PR aims to add the `alert.json` file to Vulnerability Detection E2E test report as well as the `ossec.log` and the test logs are attached. This will be useful since when the test indicates that the alert has not been found, we will be able to check if the alert has been triggered or not. For this purpose, a new function has been added to obtain the alerts from the `alert.json` file of the managers and the function in charge of collecting this information has been modified.

## Testing performed

In order to test the changes that have been made, the following environment has been released https://ci.wazuh.info/job/Wazuh_QA_environment/1025/. To avoid running the entire test and not spending so much time, only the `upgrade_package_remove_vulneraility` test has been run, which is currently expected to fail due to what is reported in this issue https://github.com/wazuh/wazuh-qa/issues/5104. When the test failed, a report was generated which indeed includes a file containing the `alert.json`, and is attached below:

image

image

[report5105.zip](https://github.com/wazuh/wazuh-qa/files/14746364/report5105.zip)

### CODE REVIEW

**Review by juliamagan — CHANGES_REQUESTED** [review]
GJ, but remember to add the changelog entry.

**Comment by MARCOSD4** [comment]
Changelog updated in https://github.com/wazuh/wazuh-qa/pull/5147/commits/ec40e06cff5179c20008d5ccb5afd4dc0b3fcb8a

**Comment by MARCOSD4** [comment]
Changelog updated in https://github.com/wazuh/wazuh-qa/pull/5147/commits/380a64151ef8257f01f59c04df9fb3121f9d01ad

File: deps/wazuh_testing/wazuh_testing/end_to_end/logs.py [hunk 1]
```diff
@@ -59,3 +59,18 @@ def get_hosts_logs(host_manager: HostManager, host_group: str = 'all') -> Dict[s
...
     return host_logs
+
+def get_hosts_alerts(host_manager: HostManager, host_group: str = 'all') -> Dict[str, str]:
+    """
+    Get the alerts in the alert.json file from the specified host group.
+
+    Parameters:
+    - host_manager (HostManager): An instance of the HostManager class for managing remote hosts.
+    - host_group (str, optional): The name of the host group where the files will be truncated.
+      Default is 'all'.
```
[the remark this answers is not in the document]
Comment (MARCOSD4) [reply deps/wazuh_testing/wazuh_testing/end_to_end/logs.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/5147/commits/99be98778053d80f263205bc05e9fb9fe46b0ccc

File: deps/wazuh_testing/wazuh_testing/end_to_end/logs.py [hunk 2]
```diff
@@ -59,3 +59,18 @@ def get_hosts_logs(host_manager: HostManager, host_group: str = 'all') -> Dict[s
         host_logs[host] = host_manager.get_file_content(host, logs_filepath_os[host_os_name])
 
     return host_logs
+
+def get_hosts_alerts(host_manager: HostManager, host_group: str = 'all') -> Dict[str, str]:
```
[the remark this answers is not in the document]
Comment (MARCOSD4) [reply deps/wazuh_testing/wazuh_testing/end_to_end/logs.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/5147/commits/99be98778053d80f263205bc05e9fb9fe46b0ccc

File: deps/wazuh_testing/wazuh_testing/end_to_end/logs.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MARCOSD4) [reply deps/wazuh_testing/wazuh_testing/end_to_end/logs.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/5147/commits/99be98778053d80f263205bc05e9fb9fe46b0ccc

File: tests/end_to_end/test_vulnerability_detector/conftest.py [hunk 3]
```diff
@@ -61,17 +61,26 @@ def collect_e2e_environment_data(test_name, host_manager) -> None:
...
-        with open(evidence_file, 'w') as evidence_file:
-            evidence_file.write(environment_logs[host])
+        host_logs_name_evidence = host + "_ossec.log"        
+        evidence_log_file = os.path.join(tests_evidences_directory, host_logs_name_evidence)
+        with open(evidence_log_file, 'w') as evidence_log_file:
+            evidence_log_file.write(environment_logs[host])
+
+    for host in environment_alerts.keys():
+        logging.info(f"Collecting alerts for {host}")
+        if host.startswith("manager"):
```
[the remark this answers is not in the document]
Comment (MARCOSD4) [reply tests/end_to_end/test_vulnerability_detector/conftest.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/5147/commits/99be98778053d80f263205bc05e9fb9fe46b0ccc

**Review by juliamagan — APPROVED** [review]

**Review by Rebits — APPROVED** [review]
LGTM

**Pull request merged**